network/src/connectivity_manager/test.rs

```diff
@@ -23,25 +23,54 @@ const MAX_TEST_CONNECTIONS: usize = 3;
 const CONNECTIVITY_CHECK_INTERVAL: Duration = Duration::from_secs(5);
 const CONNECTION_DELAY: Duration = Duration::from_millis(100);
 const MAX_CONNECTION_DELAY: Duration = Duration::from_secs(60);
+const DEFAULT_BASE_ADDR: &str = "/ip4/127.0.0.1/tcp/9090";
 
 // TODO(philiphayes): just use `CONNECTION_DELAY + MAX_CONNNECTION_DELAY_JITTER`
 // when the const adds are stabilized, instead of this weird thing...
 const MAX_DELAY_WITH_JITTER: Duration = Duration::from_millis(
     CONNECTION_DELAY.as_millis() as u64 + MAX_CONNECTION_DELAY_JITTER.as_millis() as u64,
 );
 
-fn gen_peer() -> (
-    PeerId,
-    x25519::PublicKey,
-    HashSet<x25519::PublicKey>,
-    NetworkAddress,
-) {
-    let peer_id = PeerId::random();
+fn network_address(addr_str: &'static str) -> NetworkAddress {
+    NetworkAddress::from_str(addr_str).unwrap()
+}
+fn network_address_with_pubkey(
+    addr_str: &'static str,
+    pubkey: x25519::PublicKey,
+) -> NetworkAddress {
+    network_address(addr_str).append_prod_protos(pubkey, HANDSHAKE_VERSION)
+}
+
+fn peer_id(index: usize) -> PeerId {
+    PeerId::new((index as u128).to_be_bytes())
+}
+
+fn test_peer(index: usize) -> (PeerId, Peer, x25519::PublicKey, NetworkAddress) {
+    test_peer_with_address(index, DEFAULT_BASE_ADDR)
+}
+
+fn test_peer_with_address(
+    index: usize,
+    addr_str: &'static str,
+) -> (PeerId, Peer, x25519::PublicKey, NetworkAddress) {
+    let peer_id = peer_id(index);
     let pubkey = x25519::PrivateKey::generate_for_testing().public_key();
     let pubkeys = hashset! { pubkey };
-    let addr = NetworkAddress::from_str("/ip4/127.0.0.1/tcp/9090").unwrap();
-    let addr = addr.append_prod_protos(pubkey, diem_config::config::HANDSHAKE_VERSION);
-    (peer_id, pubkey, pubkeys, addr)
+    let addr = network_address_with_pubkey(addr_str, pubkey);
+    (
+        peer_id,
+        Peer::new(vec![addr.clone()], pubkeys, PeerRole::Validator),
+        pubkey,
+        addr,
+    )
```

Comment:
so in one case you perform an op in place (Peer:new) and in other cases you define intermediate values, save tre.. err lines ;)